Assets/Scripts/LogicServices/AlgorithmService.cs

```diff
@@ -29,8 +31,15 @@ public bool Decide(Trade trade,User originator, User targetCpu)
             _target = targetCpu;
             _trade = trade;
             
-            if (_randomness.Calculate()) return !_selfBuild.Calculate(trade,targetCpu);
-            return _selfBuild.Calculate(trade,targetCpu);
+            //Decisions
+            bool randomDecision = _randomness.Calculate();
+            bool selfBuildDecision = _selfBuild.Calculate(trade, targetCpu);
+            bool buildEffectDecision = _buildEffect.Calculate(trade, targetCpu, originator);
+            
+            Debug.Log($"Random: {randomDecision} SelfBuild: {selfBuildDecision} BuildEffect: {buildEffectDecision}");
+
+            if (randomDecision) return !(selfBuildDecision && buildEffectDecision);

```

Comment:
The Randomness should be applied to a single decision, not to the whole group of decisions.